src/gtc/gtcpp/oir_to_gtcpp.py

```diff
@@ -130,7 +130,10 @@ def visit_FieldAccess(self, node: oir.FieldAccess, **kwargs: Any) -> gtcpp.Acces
         return gtcpp.AccessorRef(
             name=node.name,
             offset=self.visit(node.offset),
-            data_index=node.data_index,
+            data_index=[
```

Comment:
As mentioned using `ScalarLiteral` avoid the need for a conditional here.

Reply:
👍